### ISSUE TITLE
Incorrectly warning `Missing class import via use statement` for dynamic class instantiation.

### ISSUE THREAD

**Issue by mikebronner** [phpmd/phpmd#673]
- PHPMD version: 2.7.0
- PHP Version: 7.3.9
- Installation type: composer
- Operating System / Distribution & Version: Ubuntu 18.04

### Current Behavior
I get the following warning:
```
Missing class import via use statement (line '16', column '25').
```

for the following code:
```php
<?php

use Illuminate\Database\Seeder;

class GovernorPermissionsTableSeeder extends Seeder
{
    public function run()
    {
        $actionClass = config("genealabs-laravel-governor.models.action");
        $entityClass = config("genealabs-laravel-governor.models.entity");
        $ownershipClass = config("genealabs-laravel-governor.models.ownership");
        $permissionClass = config("genealabs-laravel-governor.models.permission");
        $roleClass = config("genealabs-laravel-governor.models.role");

        $superadmin = (new $roleClass)->whereName('SuperAdmin')->get()->first();
        $actions = (new $actionClass)->all();
        $ownership = (new $ownershipClass)->whereName('any')->get()->first();
        $entities = (new $entityClass)->all();

        foreach ($entities as $entity) {
            foreach ($actions as $action) {
                (new $permissionClass)->firstOrCreate([
                    "role_key" => $superadmin->name,
                    "action_key" => $action->name,
                    "ownership_key" => $ownership->name,
                    "entity_key" => $entity->name,
                ]);
            }
        }
    }
}
```

### Expected Behavior
There is no use of a specific class, as these are dynamic class instantiations. There is no way to import the class via a `use` statement.

### Steps To Reproduce:
Run PHPMD against the above code.

## Checks before submitting
* [X] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [X] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [X] I have added every step to reproduce the bug.
* [X] If possible I added relevant code examples.
* [X] This issue is about 1 bug and nothing more.
* [X] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by tvbeek** [comment]
@mikebronner thanks for your issue. It is an interesting one, I think the rule need to skip variable classes because there is no way to tell if we can find it or not.

**Comment by kylekatarnls** [comment]
Yes, I will fix that with a `substr($className, 0, 1) !== '$'` I don't think we should do anything more complex than that.
